static/app/views/ddm/metricsExplorer.tsx

```diff
@@ -178,19 +120,42 @@ function QueryBuilder({setQuery}: QueryBuilderProps) {
                   </Fragment>
                 ),
               }))}
-            value={state.mri}
+            value={metricsQuery.mri}
             onChange={option => {
-              dispatch({type: 'mri', value: option.value});
+              const availableOps = meta[option.value]?.operations.filter(isAllowedOp);
+              const selectedOp = availableOps.includes(metricsQuery.op ?? '')
+                ? metricsQuery.op
+                : availableOps[0];
+              router.push({
+                ...router.location,
+                query: {
+                  ...router.location.query,
+                  mri: option.value,
+                  op: selectedOp,
+                  groupBy: undefined,
+                },
+              });
```

Comment:
nit: this pattern is repeated quite often in this component. Does it make sense to create a helper for that?
```ts
const updateQuery = (partialQuery: Record<string, any>) => {
  router.push({
    ...router.location,
    query: {
      ...router.location.query,
      ...partialQuery,
    },
  })
}
```

Reply:
absolutely! done ❤️